tools/maven/suppressions-core.xml

```diff
@@ -121,4 +121,7 @@ under the License.
 	<suppress
 		files="(.*)test[/\\](.*)testutils[/\\](.*)"
 		checks="RedundantModifier|JavadocParagraph|JavadocType|JavadocStyle|StaticVariableNameCheck|LocalFinalVariableName|EmptyLineSeparator"/>
+
+	<!-- suppress check java 17 tests -->
+	<suppress files="(.*)test[/\\].*[/\\]Java17.*.java" checks=".*"/>
```

Comment:
When looking for the reason for this weird checkstyle rule, I found https://github.com/checkstyle/checkstyle/issues/8023. I verified that this is actual an issue in our current setup.

It's not an issue for this PR because we don't need this feature of records. But I'm wondering whether we should start collecting reasons like that to initiate a discussion on upgrading the checkstyle version. 🤔